### PR TITLE
fix: use rspack config chain

### PR DESCRIPTION
**Pull request by bvanjoi**
Using the webpack type in `﻿webpack-chain` is not an optimal choice for bundlers based on rspack.

### CODE REVIEW

**Comment by bvanjoi** [comment]
related to https://github.com/web-infra-dev/rspack/issues/5045

File: packages/compat/webpack/tests/plugins/__snapshots__/default.test.ts.snap [hunk 1]
```diff
@@ -2,16 +2,7 @@
...
-  "cache": {
-    "buildDependencies": {
-      "packageJson": [
-        "<ROOT>/package.json",
-      ],
-    },
-    "cacheDirectory": "<ROOT>/node_modules/.cache/webpack",
-    "name": "web-development",
-    "type": "filesystem",
-  },
```
Comment (9aoy) [inline packages/compat/webpack/tests/plugins/__snapshots__/default.test.ts.snap:14]: snapshot need update

File: packages/shared/src/chain.ts [hunk 2]
```diff
@@ -257,7 +257,7 @@ export function applyScriptCondition({
 }) {
   // compile all folders in app directory, exclude node_modules
   rule.include.add({
-    and: [context.rootPath, { not: NODE_MODULES_REGEX }],
```
Comment (9aoy) [inline packages/shared/src/chain.ts:260]: Webpack supports the RegExp type, and rspack also previously supported the RegExp type. Isn't it supported now?
image

Comment (bvanjoi) [reply packages/shared/src/chain.ts:260]: see https://github.com/web-infra-dev/rspack/issues/5057

Comment (9aoy) [reply packages/shared/src/chain.ts:260]: we can revert this change~

**Review by 9aoy — APPROVED** [review]

File: packages/shared/compiled/webpack-chain/types/index.d.ts [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
   Configuration,
   Compiler,
-  WebpackPluginInstance,
   RuleSetRule,
-  ResolveOptions,
-} from 'webpack';
+  ResolveOptions
+} from '@rspack/core';
+
+declare interface WebpackPluginInstance {
```
Comment (chenjiahan) [inline packages/shared/compiled/webpack-chain/types/index.d.ts:8]: We should not modify the compiled types directly, it is generated by `ncc`.

**Pull request merged**